props/saber_BC_buttons.h

```diff
@@ -386,31 +1134,321 @@ EFFECT(volmax);     // for maximum volume reached
 #warning The faston.wav sound will be replaced with fastout.wav. If you have a good reason to keep faston.wav as is, please post at https://crucible.hubbe.net/
 EFFECT(faston);     // for EFFECT_FAST_ON. Being replaced by fastout.wav, which is already defined in the main OS.
 #endif
-EFFECT(blstbgn);    // for Begin Multi-Blast
-EFFECT(blstend);    // for End Multi-Blast
 EFFECT(push);       // for Force Push gesture
-EFFECT(monosfx);    // for Monophonically played sounds (iceblade, seismic charge etc...)
-EFFECT(swap);       // for standalone triggering EffectSequence<>
+EFFECT(tr);         // for EFFECT_TRANSITION_SOUND, use with User Effects.
+EFFECT2(trloop, trloop);  // have YET to play with this
 EFFECT(mute);       // Notification before muted ignition to avoid confusion.
+EFFECT(mzoom);      // for Spam Blast enter/exit
+
+#ifndef BC_ENABLE_OS_MENU
+template<class SPEC>
+struct BCVolumeMode : public SPEC::MenuBase {
+  const int max_volume_ = VOLUME;
+  const int min_volume_ = VOLUME * 0.10;
+  float initial_volume_ = 0.0;
+  int initial_percentage_ = 0;
+  int last_volume_ = 0;
+  int current_volume_ = 0;
+  int percentage_ = 0;
+
+int steps_per_revolution() override {
+  return 12;  // adjust for sensitivity
+}
+  uint16_t size() override {
+    return 10;
+  }
 
-class DelayTimer {
-public:
-    DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
+  void mode_activate(bool onreturn) override {
+    PVLOG_NORMAL << "** Enter Volume Menu\n";
+    initial_volume_ = dynamic_mixer.get_volume();
+    initial_percentage_ = round((initial_volume_ / max_volume_) * 10) * 10;
+    last_volume_ = initial_volume_;
+    current_volume_ = initial_volume_;
```

Comment:
Isn't current_volume_ always the same as dynamic_mixer.get_volume()? (If so, why have it?)

Reply:
I figured using the variable would reduce calls to dynamic_mixer.get_volume(), as well as make it a bit more readable? like in comparisons of `if (current_volume_ < max_volume_)`

Reply:
Actually, it's last_volume_ that's the superfluous thing. It was used in a previous iteration, but could go away now. Removing.

Reply:
but isn't current_volume_ *also* superfluous?

Reply:
> I figured using the variable would reduce calls to dynamic_mixer.get_volume()

Pretty sure the compiler will figure that out and make it equally efficient as accessing a variable.

> as well as make it a bit more readable? 

It is true that the code can become more readable, but the way to do that is to use *local* variables. (variables which only exists within one function.) When you persist the variable between functions, the chance of making a mistake somewhere so that current_volume_ and dynamic_mixer.get_volume() are *different* becomes significant, and checking for those sort of problems adds to the time it takes to read the code.

Generally speaking, avoiding storing the same thing in two places is one of the fundamental principles of good code.

Reply:
ok. moved current_volume_ inside functions and only where useful.